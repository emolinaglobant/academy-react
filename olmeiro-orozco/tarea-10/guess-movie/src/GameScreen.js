import Proptypes from "prop-types";

import { CentralGame } from "./components/CentralGame";

export const GameScreen = ({ lives, points, numberQuestion }) => {
  return (
    <>
      <CentralGame lives={lives} points={points} question={numberQuestion} />
    </>
  );
};

GameScreen.prototypes = {
  lives: Proptypes.number.isRequired,
  points: Proptypes.number.isRequired,
  numberQuestion: Proptypes.number.isRequired,
};

GameScreen.defaultProps = {
  lives: 3,
  points: 0,
  numberQuestion: 0,
};
