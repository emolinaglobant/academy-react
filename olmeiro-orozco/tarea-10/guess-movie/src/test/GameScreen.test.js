import React from "react";
import '@testing-library/jest-dom'

import { shallow } from "enzyme";
import { GameScreen } from '../GameScreen';

describe('Tests component GameScreen', () => { 

 const lives= 3;
 const points= 0;
 const numberQuestion= 0;


  test("should render component correctly", () => {
    const wrapper = shallow(
      <GameScreen lives={lives} points={points} question={numberQuestion} />
    );
    expect(wrapper).toMatchSnapshot();
  });
 })
