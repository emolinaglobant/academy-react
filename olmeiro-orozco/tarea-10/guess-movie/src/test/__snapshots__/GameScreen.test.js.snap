// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests component GameScreen should render component correctly 1`] = `
<Fragment>
  <CentralGame
    lives={3}
    points={0}
    question={0}
  />
</Fragment>
`;
